inc/endpoints/class-components-registry-endpoint.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Class file for Components Registry endpoint.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving\REST_API;
+
+use WP_Irving\REST_API\Endpoint;
+
+/**
+ * Components Registry endpoint.
+ */
+class Components_Registry_Endpoint extends Endpoint {
+
+	/**
+	 * Register the REST API routes.
+	 */
+	public function register_rest_routes() {
+		register_rest_route(
+			self::get_namespace(),
+			'/registered-components/',
+			[
+				'methods'  => \WP_REST_Server::READABLE,
+				'callback' => function() {
+					return \WP_Irving\Components\get_registry()->get_registered_components();
+				},
+			]
+		);
+	}
+}
+
+new Components_Registry_Endpoint();
```

Comment:
Don't instantiate the class when the file is included. This should happen somewhere else. I'd like for us to follow the [PSR-1 standard](php-fig.org/psr/psr-1/) here:

```
Files SHOULD either declare symbols (classes, functions, constants, etc.) or cause side-effects (e.g. generate output, change .ini settings, etc.) but SHOULD NOT do both.
```